fe/fe-core/src/main/java/com/starrocks/statistic/AnalyzeManager.java

```diff
@@ -263,6 +271,102 @@ public void clearStatisticFromDroppedTable() {
         dropHistogramStatsMetaAndData(statsConnectCtx, tableIdHasDeleted);
     }
 
+    public void dropPartition(long partitionId) {
+        dropPartitionIds.add(partitionId);
+    }
+
+    public void clearStatisticFromDroppedPartition() {
+        checkAndDropPartitionStatistics();
+        dropPartitionStatistics();
+    }
+
+    private void dropPartitionStatistics() {
+        if (dropPartitionIds.isEmpty()) {
+            return;
+        }
+
+        ConnectContext statsConnectCtx = StatisticUtils.buildConnectContext();
+        statsConnectCtx.setThreadLocalInfo();
+
+        List<Long> pids = dropPartitionIds.stream().limit(Config.expr_children_limit / 2).collect(Collectors.toList());
+
+        StatisticExecutor executor = new StatisticExecutor();
+        statsConnectCtx.setThreadLocalInfo();
+        if (executor.dropPartitionStatistics(statsConnectCtx, pids)) {
+            pids.forEach(dropPartitionIds::remove);
+        }
+    }
+
+    private void checkAndDropPartitionStatistics() {
+        if (!Config.statistic_check_expire_partition || checkTableIds.isEmpty()) {
+            return;
+        }
+
+        if (checkTableIds.contains(CHECK_ALL_TABLES)) {
+            checkTableIds.clear();
+            List<Long> dbIds = GlobalStateMgr.getCurrentState().getDbIds();
+            for (Long dbId : dbIds) {
+                Database db = GlobalStateMgr.getCurrentState().getDb(dbId);
+                if (null == db || StatisticUtils.statisticDatabaseBlackListCheck(db.getFullName())) {
+                    continue;
+                }
+
+                for (Table table : db.getTables()) {
+                    if (table == null || !(table.isOlapOrCloudNativeTable() || table.isMaterializedView())) {
+                        continue;
+                    }
+                    checkTableIds.add(new Pair<>(dbId, table.getId()));
+                }
+            }
+
+        }
+
+        List<Pair<Long, Long>> checkDbTableIds = Lists.newArrayList();
+        List<Long> checkPartitionIds = Lists.newArrayList();
+        ConnectContext statsConnectCtx = StatisticUtils.buildConnectContext();
+
+        int exprLimit = Config.expr_children_limit / 2;
+        for (Pair<Long, Long> dbTableId : checkTableIds) {
+            Database db = GlobalStateMgr.getCurrentState().getDb(dbTableId.first);
+            if (null == db) {
+                continue;
+            }
+
+            Table table = db.getTable(dbTableId.second);
+            if (table == null) {
+                continue;
+            }
+
+            List<Long> pids = table.getPartitions().stream().map(Partition::getId).collect(Collectors.toList());
+
+            if (pids.size() > exprLimit) {
```

Comment:
please add some comments to descirbe this strategy.

Reply:
OK，just for pass the parse check